bears/general/FilenameBear.py

```diff
@@ -74,12 +83,44 @@ def run(self, filename, file,
         if ignore_uppercase_filenames and filename_without_extension.isupper():
             return
 
-        if new_name != filename_without_extension:
-            diff = Diff(file, rename=os.path.join(head, new_name + extension))
+        if (new_name != filename_without_extension or
+            not filename_without_extension.startswith(filename_prefix) or
+                not filename_without_extension.endswith(filename_suffix)):
+            correct_prefix_name = (new_name
+                                   if
+                                   filename_without_extension.startswith(
+                                                               filename_prefix)
+                                   else
+                                   filename_prefix + new_name)
+
+            correct_prefixsuffix_name = (correct_prefix_name
+                                         if
+                                         filename_without_extension.endswith(
+                                                               filename_suffix)
+                                         else
+                                         correct_prefix_name + filename_suffix)
+
+            diff = Diff(file, rename=os.path.join(
+                head, correct_prefixsuffix_name + extension))
+
+            FilenameBear_messages = [
+                'Filename does not follow {} naming-convention.'.format(
+                        file_naming_convention),
+                'Filename does not use the {} prefix.'.format(
+                        filename_prefix),
+                'Filename does not use the {} suffix.'.format(
+                        filename_suffix)]
+
+            selectors = [
+                        new_name != filename_without_extension,
+                        not filename_without_extension.startswith(
+                            filename_prefix),
+                        not filename_without_extension.endswith(
+                            filename_suffix)]
 
             yield Result(
                 self,
-                'Filename does not follow {} naming-convention.'.format(
-                    file_naming_convention),
+                '\n'.join(
+                    itertools.compress(FilenameBear_messages, selectors)),
```

Comment:
Oh maybe we can be super smart and have an exception if there's only a single message:
```
if len(messages) == 1:
    # no bullet, just the message
else:
    # use bullets together with messages
```